contracts/MarketWithRisk.sol

```diff
@@ -39,18 +49,29 @@ contract MarketWithRisk is Market {
 		uint256 odds,
 		uint256 close,
 		uint256 end,
-		uint256 risk
+		uint256 risk,
 		SignatureLib.Signature calldata signature
 	) external returns (uint256) {
-		return _back(
+		bytes32 messageHash = keccak256(abi.encodePacked(
 			nonce,
 			propositionId,
 			marketId,
-			wager,
 			odds,
 			close,
 			end,
-			signature
+			risk
+		));
+
+		require(isValidSignature(messageHash, signature) == true, "back: Invalid signature");
+
+		uint256 payout = wager * _getOddsWithRisk(wager, odds, propositionId, marketId, risk);
+		return _back(
```

Comment:
could we just override the `back` function in this contract to use risk?

Reply:
No because we have an extra param `risk`